api/web3server.go

```diff
@@ -29,16 +28,46 @@ import (
 )
 
 const (
-	_contentType                 = "application/json"
 	_metamaskBalanceContractAddr = "io1k8uw2hrlvnfq8s2qpwwc24ws2ru54heenx8chr"
 )
 
-// Web3Server contains web3 server and the pointer to api coreservice
-type Web3Server struct {
-	queryLimit  uint64
-	web3Server  *http.Server
-	coreService CoreService
-	cache       apiCache
+type (
+	// Web3Handler handle JRPC request
+	Web3Handler interface {
+		HandlePOSTReq(reader io.Reader) interface{}
+	}
+
+	// Web3Server contains web3 server and the pointer to api coreservice
+	Web3Server struct {
+		coreService  CoreService
+		cache        apiCache
+		httpSvr      *HTTPServer
+		websocketSvr *WebsocketServer
```

Comment:
We may move httpServer and webSocketServer into ServerV2, such that
* `ServerV2`: start http handle all incoming requests and dispatch to corresponding component
* `Web3Server`: handle the web3 request
* `GrpcServer`: handle the grpc request